compatibility/sandbox-migration/runner/Migration/KeyTransfer.hs

```diff
@@ -0,0 +1,157 @@
+-- Copyright (c) 2020 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+-- SPDX-License-Identifier: Apache-2.0
+
+{-# LANGUAGE DerivingStrategies #-}
+{-# LANGUAGE DuplicateRecordFields #-}
+module Migration.KeyTransfer (test) where
+
+import Control.Monad
+import Control.Monad.Except
+import qualified Data.Aeson as A
+import Data.List
+import qualified Data.Text as T
+import GHC.Generics (Generic)
+import System.IO.Extra
+import System.Process
+
+import Migration.Types
+
+diff :: Eq a => [a] -> [a] -> [a]
+diff left right = (left \\ right) ++ (right \\ left)
+
+equivalent :: Eq a => [a] -> [a] -> Bool
+equivalent xs = null . (diff xs)
+
+--data Result = Result
+--  { oldKeptAssets :: [Tuple2 (ContractId Asset) Asset]
+--  , newKeptAssets :: [Tuple2 (ContractId Asset) Asset]
+--  , oldTransferredAssets :: [Tuple2 (ContractId Asset) Asset]
+--  , newTransferredAssets :: [Tuple2 (ContractId Asset) Asset]
+--  , oldTransactions :: [Transaction]
+--  , newTransactions :: [Transaction]
+--  } deriving Generic
+test :: FilePath -> FilePath -> Test ([Tuple2 (ContractId Asset) Asset], [Transaction]) Result
+test step modelDar = Test {..}
+  where
+    initialState = ([], [])
+    executeStep sdkVersion host port _state = withTempFile $ \outputFile -> do
+        let suffix = getSdkVersion sdkVersion
+        callProcess step
+            [ "--host=" <> host
+            , "--port=" <> show port
+            , "--output", outputFile
+            , "--dar=" <> modelDar
+            , "--test=key-transfer," <> T.unpack (getParty testOwner) <> "," <> T.unpack (getParty testReceiver) <> "," <> suffix
+            ]
+        either fail pure =<< A.eitherDecodeFileStrict' outputFile
+    validateStep sdkVersion (prevAssets, prevTransactions) Result {..} = do
+        let suffix = getSdkVersion sdkVersion
+        unless (equivalent oldAssets prevAssets) $
+            throwError ("The old assets do not match those returned by the previous run: " <> show oldAssets)
+        unless (equivalent oldTransactions prevTransactions) $
```

Comment:
Are you deliberately ignoring the order here? For the ACS that seems reasonable but I thought the transaction stream was ordered.

Reply:
Addressed by https://github.com/digital-asset/daml/pull/6298/commits/e0e6f87047d1ec630cc646660420be4eacaafe40